src/Tasks/Microsoft.NET.Build.Tasks/ClsidMap.cs

```diff
@@ -189,12 +187,46 @@ private static string GetTypeGuid(MetadataReader reader, TypeDefinition type)
                 if (reader.StringComparer.Equals(attributeType.Namespace, "System.Runtime.InteropServices") && reader.StringComparer.Equals(attributeType.Name, "GuidAttribute"))
                 {
                     CustomAttributeValue<KnownType> data = attribute.DecodeValue(new TypeResolver());
-                    return (string)data.FixedArguments[0].Value;
+                    return $"{{{(string)data.FixedArguments[0].Value}}}";
                 }
             }
             throw new BuildErrorException(Strings.ClsidMapExportedTypesRequireExplicitGuid, GetTypeName(reader, type));
         }
 
+        private static void WriteClsidMap(Stream clsidMapFile, Dictionary<string, ClsidEntry> map)
+        {
+            const int SyncWriteThreshold = 1_000; // Copy to stream after writing this many bytes
+
+            ArrayBufferWriter buffer = new ArrayBufferWriter();
+            Utf8JsonWriter json = new Utf8JsonWriter(buffer);
+
+            long prevBytesWritten = 0;
+
+            // Write some JSON, let's say an array of JSON objects in a loop
+            json.WriteStartObject();
+
+            foreach (KeyValuePair<string, ClsidEntry> clsid in map)
+            {
+                json.WriteStartObject(clsid.Key);
+
+                json.WriteString("assembly", clsid.Value.Assembly);
+                json.WriteString("type", clsid.Value.Type);
+
+                json.WriteEndObject();
+
+                prevBytesWritten = json.BytesWritten - prevBytesWritten;
+                if (prevBytesWritten > SyncWriteThreshold)
+                {
+                    json.Flush(isFinalBlock: false);
+                    buffer.CopyTo(clsidMapFile);
```

Comment:
I believe there is bug in the sample which will cause issues here.

After flushing, you need to reset the json writer local by re-instantiating it.

If you are keeping this `SyncWriteThreshold` pattern here (which in general should not be needed), change it to the following.

Add:
```C#

            foreach (KeyValuePair<string, ClsidEntry> clsid in map)
            {
                json.WriteStartObject(clsid.Key);

                json.WriteString("assembly", clsid.Value.Assembly);
                json.WriteString("type", clsid.Value.Type);

                json.WriteEndObject();

                if (json.BytesWritten > SyncWriteThreshold)
                {
                    json.Flush(isFinalBlock: false);
                    buffer.CopyTo(clsidMapFile);
                    json = new Utf8JsonWriter(buffer, json.GetCurrentState());
                }
            }
```

I'll fix the sample docs and call out that this pattern is mainly useful for async writes, not synchronous ones. Thanks!

Reply:
Btw, feedback/contribution to the porting guide would be most welcome so it can be improved :)

https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/porting_guide/README.md